enumerable_methods.rb

```diff
@@ -0,0 +1,213 @@
+      end
+    elsif arg[0] == Integer || arg[0] == Float || arg[0] == String
+      my_each do |el|
+        i += 1 if el.class != arg[0]
+      end
+    else
+      my_each do |el|
+        i += 1 if el != arg[0]
+      end
+    end
+    i == size
+  end
+
+  def my_count(*arg)
+    counter = 0
+    if block_given?
+      my_each do |el|
+        counter += 1 if yield(el) == true
+      end
+      counter
+    elsif arg.empty?
+      size
+    else
+      my_each do |el|
+        counter += 1 if el == arg[0]
+      end
+      counter
+    end
+  end
+end
+
+module Enumerable
+  def my_map(*_arg)
+    return 'No block given' unless block_given?
+
+    i = 0
+    arr = []
+    my_each do |el|
+      arr[i] = yield(el)
+      i += 1
+    end
+    arr
+  end
+
+  def my_inject(num = nil)
+    sum = 0
+    if block_given?
+      my_each do |el|
+        sum = yield(sum, el)
+      end
+      return sum if num.nil?
+
+      sum + num
+    elsif num.class == Symbol || num.class == String
+      sum = nil
+      my_each { |i| sum = sum.nil? ? i : sum.send(num, i) }
+      sum
+    else
+      my_each do |el|
+        sum += el
+      end
+      return sum if num.nil?
+
+      sum + num
+    end
+  end
+
+  def multiply_els(arr)
+    arr.my_inject(:*)
+  end
+end
+
+p 'My_each method:'
+p([nil, 2.5, 'hi', -7, true].my_each { |el| el })
+puts
+p 'My_each_with_index:'
+[nil, 2.5, 'hi', -7, true].my_each_with_index { |el, i| p("Element: #{el}, Index: #{i}") }
+puts
+p 'My_select method:'
+[1, 2, 3, 4, 5, 6].my_select { |n| n > 3 }
+puts
+p 'My_all? method:'
+p 'Using block:'
+p(%w[ant ant ant].my_all? { |word| word.length >= 2 })
+p 'Using Class:'
+p(%w[ant ant ant].my_all?(String))
+p 'Testing cases:'
+p(%w[ant ant ant].my_all?('ant'))
+p(%w[ant ant bear].my_all?('ant'))
+p(%w[bear cat dog].my_all?('ants'))
+p([3, 5, 2].my_all?(Integer))
+puts
+p 'My_any? method:'
+p 'Using block:'
+p(%w[ant ant ant].my_any? { |word| word.length >= 2 })
+p 'Using Class:'
+p(%w[ant ant ant].my_any?(String))
+p 'Testing cases:'
+p(%w[ant ant ant].my_any?('ant'))
+p(%w[ant ant bear].my_any?('ant'))
+p(%w[bear cat dog].my_any?('ants'))
+
+puts
+p 'My_none? method:'
+p 'Using block:'
+p(%w[ant ant ant].my_none? { |word| word.length >= 2 })
+p 'Using Class:'
+p(%w[ant ant ant].my_none?(Integer))
+p 'Testing cases:'
+p(%w[ant ant ant].my_none?('ant'))
+p(%w[ant ant bear].my_none?('ant'))
+p(%w[bear cat dog].my_none?('ants'))
+p([3, 5, 2].my_none?(Integer))
+puts
+p 'My_count method:'
+p 'Using argument:'
+p([1, 2, 4, 23, 34, 143, 143, 143].my_count(143))
+p 'Using block:'
+p([1, 2, 4, 23, 34, 143, 143, 143].my_count { |n| n > 20 })
+p 'Without argument nor block:'
+p([1, 2, 4, 23, 34, 143, 143, 143].my_count)
+puts
+p 'My_map method:'
+p 'Using block:'
+p([1, 2, 3, 22, 5].my_map { |x| x * 5 })
+p 'Using proc:'
+proc1 = proc { |x| x**2 }
+p [1, 2, 3, 22, 5].my_map(&proc1)
+puts
+p 'My_inject method:'
+p 'Using argument:'
+p [1, 2, 3, 2, 5].my_inject(5)
```

Comment:
Kindly fix the inject method, when compared it with the custom `inject` method, we get a (TypeError) `inject': 5 is not a symbol nor a string (TypeError)`

Reply:
- Read me updated
- Creating a new branch to just paste the same code would be redundant no?
- condition for my_inject corrected.
- About the linters, as said, the first 2 ones are because they want to force me into using each (which is not allowed) and the other issues are because of the complexity of if/else statements, that I guess it is not allowed to break the methods as well.